c/src/CMakeLists.txt

```diff
@@ -16,9 +18,25 @@ swift_add_library(sbp
 )
 add_library(swiftnav::sbp ALIAS sbp)
 
-target_include_directories(sbp PUBLIC ${PROJECT_SOURCE_DIR}/include PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include)
-set_target_properties(sbp PROPERTIES
-        POSITION_INDEPENDENT_CODE ON)
+generate_export_header(sbp
+  DEPRECATED_MACRO_NAME CMAKE_SBP_DEPRECATED
+  NO_DEPRECATED_MACRO_NAME CMAKE_SBP_NO_DEPRECATED
+  INCLUDE_GUARD_NAME LIBSBP_SBP_EXPORT_H
+)
+target_include_directories(sbp
+  PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_LIBDIR}>
+  PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}/include
+)
+set_target_properties(sbp
+  PROPERTIES
+    VISIBILITY_INLINES_HIDDEN true
```

Comment:
these options make it so that all symbols on Linux OS are private, in the same way they are on Windows Visual studio. This is done to:

1. minimize the shared library size by not exposing all symbols
2. making it easier to pick up error on Linux/Mac machines that would appear on Windows